src/ccnl-core/src/ccnl-prefix.c

```diff
@@ -578,62 +578,49 @@ ccnl_prefix_to_path_detailed(struct ccnl_prefix_s *pr, int ccntlv_skip,
     else
         buf = prefix_buf2;
     */
-    char *buf = (char*) ccnl_malloc(CCNL_MAX_PREFIX_SIZE);
+    char *buf = (char*) ccnl_malloc(CCNL_MAX_PREFIX_SIZE+1);
     if (buf == NULL) {
         DEBUGMSG_CUTL(ERROR, "ccnl_prefix_to_path_detailed: malloc failed, exiting\n");
         return NULL;
     }
-
+    memset(buf, 0, CCNL_MAX_PREFIX_SIZE+1);
     return ccnl_prefix_to_str_detailed(pr, ccntlv_skip, escape_components, call_slash, buf, CCNL_MAX_PREFIX_SIZE);
 }
 
 char*
 ccnl_prefix_to_str_detailed(struct ccnl_prefix_s *pr, int ccntlv_skip, int escape_components, int call_slash,
-                            char *buf, int buflen) {
-    int len = 0, i, j;
+                            char *buf, size_t buflen) {
+    size_t len = 0, i, j;
     (void)i;
     (void)j;
     (void)len;
     (void) call_slash;
-    (void)buflen; //fixme, prevent bufferoverrun here!
 #ifdef USE_NFN
-    // len += sprintf(buf + len, "cmpcnt: %i|", pr->compcnt);
     if (pr->nfnflags & CCNL_PREFIX_NFN) {
-        len += sprintf(buf + len, "nfn");
+        len += snprintf(buf + len, buflen - len, "nfn");
+        if (buflen < len){
```

Comment:
I don't want be  a dick, but I've thought more about something like
```C
int result = snprintf(buf + len, buflen - len, "nfn");

if (!(result > -1 && result < (buflen - len))) {
     /* you probably want some log debug statement here */
     return NULL;
}
```
So you check a) if an error occurred and b) if the intended number of bytes have been written. Also, you can't check for errors with ``size_t`` since this type is unsigned. You could use ``ssize_t`` (which is signed) but honestly, I don't know if we should distinguish between the size (which should be unsigned) and a return value (which can be -1 for errors, but also size).

Reply:
I agree, i quickly checked the doc, snprintf returns an int, so I will use int for result.